### ISSUE TITLE
i18 Support

### ISSUE THREAD

**Issue by TianMengLucky** [Kesomannen/gale#50]
I'm curious about whether Gale will support other languages, as I learned about Gale from https://github.com/ebkr/r2modmanPlus/issues/1305#issuecomment-2241042268 and language is a hindrance to my use of the mod manager, but the r2modman is just too slow to work efficiently, and https://github.com/ebkr/r2modmanPlus/pull/1367's PR about adding other languages has been stuck for many months, so I'm curious if Gale will support other languages?

**Comment by Kesomannen** [comment]
I haven't looked into it, but it honestly seems like a pain to implement. Then you also have to get and organize people to do the translations, So idk, not on the top of the list right now

**Comment by TianMengLucky** [comment]
> I haven't looked into it, but it honestly seems like a pain to implement. Then you also have to get and organize people to do the translations, So idk, not on the top of the list right now

Translation is not a big problem, the main problem is that there is no i18 function

**Comment by Kesomannen** [comment]
Yeah, but setting it up and going through everywhere replacing strings is quite a lot of work. Of course if you want to do it I'm open to that. It's just personally I don't see it worth while right now, I'm only one person behind this whole thing 😅

**Comment by Serial-ATA** [comment]
I can work on this over the weekend. If translations are gonna be supported it's best to get it done sooner rather than later.

**Comment by TianMengLucky** [comment]
I can help replace strings, but I don't know SvelteKit and don't know how to implement i18n functionality

**Comment by Serial-ATA** [comment]
I've already implemented it, I'm just in the process of replacing strings which I'll continue over the weekend.

**Comment by TianMengLucky** [comment]
Github repository I will translate Simplified Chinese for i18n when it is updated.

**Comment by Damglador** [comment]
> Then you also have to get and organize people to do the translations, So idk, not on the top of the list right now
You can set up a Crowdin project for translations and just link it in "Help" button, so everyone can contribute to their language. Crowdin is a freemium service, but it has a plan for Open Source projects that's better than the free one.

**Comment by Damglador** [comment]
I don't think that localization is needed for mod managers, but that would be a nice thing to have, and another plus compared to r2modman

**Comment by TianMengLucky** [comment]
> I don't think that localization is needed for mod managers, but that would be a nice thing to have, and another plus compared to r2modman

I think localisation is something that should be done for every user-facing service app, not being able to use an app in your native language is a pain!

**Comment by SummonHIM** [comment]
Im starting i18n this project.
https://github.com/SummonHIM/gale/commit/c104421f8258b802478283fa3e136545928e27e3
When everything done, will send a pr to upstream.

but about the tauri part. im not familiar with rust. so  i will only translate the js part.

**Comment by TianMengLucky** [comment]
> Im starting i18n this project. [SummonHIM@c104421](https://github.com/SummonHIM/gale/commit/c104421f8258b802478283fa3e136545928e27e3) When everything done, will send a pr to upstream.
> 
> but about the tauri part. im not familiar with rust. so i will only translate the js part.

Rust-related content should not need to be translated

**Comment by TianMengLucky** [comment]
> Im starting i18n this project. [SummonHIM@c104421](https://github.com/SummonHIM/gale/commit/c104421f8258b802478283fa3e136545928e27e3) When everything done, will send a pr to upstream.
> 
> but about the tauri part. im not familiar with rust. so i will only translate the js part.

You can open a PR first and wait for it to complete before merging it

**Comment by TianMengLucky** [comment]
@SummonHIM  统一在这里进行吧 https://github.com/Kesomannen/gale/pull/90

**Comment by SummonHIM** [comment]
> @SummonHIM 统一在这里进行吧 #90

昨晚没看到不好意思。还是等我把语言切换按钮挫出来再发pr吧。能否把你的更改先发到我那？

sry for not replying such a long time. ill send pr to upstream when i finish the language menu. Its not the right time yet. u can send your changes as pr to me?

建议键名不要太长，不要超过五个单词。比如说`"Enter a name for the new profile": "输入新配置名",`。
可以用`"Enter profile name description"`代替

**Comment by lostone-zc** [comment]
Also a feature related to i18n, support search/show game name in other languages. ~~(Although searching can be solved by adding other languages name in aka(alias) part)~~

**Comment by Kesomannen** [comment]
What happened to this? Why did you close the PR?

**Comment by TianMengLucky** [comment]
> What happened to this? Why did you close the PR?

Both he and I are busy with other things and can't continue to finish it for the time being, we will reopen it later if we have free time

**Comment by TianMengLucky** [comment]
#103

**Comment by TianMengLucky** [comment]
#434
